en/computer_vision/collision_prevention.md

```diff
@@ -2,78 +2,92 @@
+> **Note** If there is no sensor data in a particular direction, velocity in that direction is restricted to 0 (preventing the vehicle from crashing into unseen objects).
 
-The data from all sensors is fused into a range scan (a 360 degree map of the sensor data/state from around the vehicle), which is then input to the collision prevention algorithm.
+Delay, both in the vehicle tracking velocity setpoints and in receiving sensor data from external sources, is conservatively estimated via the [MPC_COL_PREV_DLY](#MPC_COL_PREV_DLY) parameter.
+This should be [tuned](#delay_tuning) to the specific vehicle.
 
-The collision prevention algorithm only uses part of the range scan data in the calculation of the velocity limit (which we refer to as the *considered sensor data*). 
-This comprises the sensor data from the range scan that is within two [MPC_COL_PREV_ANG](../advanced_config/parameter_reference.md#MPC_COL_PREV_ANG) degree arcs centered around the commanded direction.
+If the sectors adjacent to the commanded sectors are 'better' by a significant margin, the direction of the requested input can be modified by up to the angle specified in [MPC_COL_PREV_CNG](#MPC_COL_PREV_CNG).
+This helps to fine-tune user input to 'guide' the vehicle around obstacles rather than getting stuck against them.
 
-Provided none of the *considered sensor data* includes data below the minimum allowed distance, the executed speed depends on the geometry, considered data, and actual sensor values.
-The vehicle will not move if **any** *considered sensor data* is below the minimum allowed separation.
 
-> **Tip** Generally 45 degrees is a good compromise value. 
-  Using a larger angle reduces the chance of clipping obstacles, but can make it feel like the vehicle is "always getting "stuck" (as you're more likely to detect an obstacle within the minimum distance).
+### MPC_COL_PREV_DLY Delay Tuning {#delay_tuning}
 
-The diagrams below are used to illustrate how the setting works, where:
-- The minimum distance (separation) `MPC_COL_PREV_D = 4m`, and the sensor range is 12m.
-- The pink line shows where the user is trying to move the vehicle (this is the same for both diagrams).
-- The blue line shows the angle where received sensor data *can be used* (and what will be discarded).
-  - This is equal to double `MPC_COL_PREV_ANG` degrees centred on the commanded setpoint.
-  - On the left diagram the `MPC_COL_PREV_D` angle is 90 degrees, so all sensor data from the right hand side of the blue line is included.
-  - On the right the angle is 45 degrees, resulting in a smaller set of sensor data being used (note the sensor data close to the obstacle is excluded!)
-- The red arc shows the angle where sensor data *will be used*.
-  This is the subset of sensors that are both in range of an obstacle, and within the angle defined by the blue line.
+There are two main sources of delay which should be accounted for: *sensor delay*, and vehicle *velocity setpoint tracking delay*.
```

Comment:
How do you calculate the sensor delay/What if you have both a sensor and a setpoint tracking delay. 

I would guess that this is all calculated using the logs as indicated, but that isn't clear.

Reply:
Characterizing sensor delay is a huge pain. We did it by overlaying a known-low-latency sensor onto our unknown sensor, and then applying movements. I'm not sure if there is a better way.

Reply:
@jkflying Thanks, but I have no idea what that means. I would have thought that since there is only one parameter the tracking delay must incorporate the sensor delay?

Anyway, I will merge this, but am assuming you will come back and clarify :-)

Reply:
Yes, the single parameter incorporates both delays, you just add them together.

Reply:
Shouldn't the measured tracking error incorporate the sensor delay by default? (ie you get the measured velocity delay in part due to the sensor delay?). Then you'd be adding a delay to something that is already "factored in".

@jkflying I could be being stupid here  ...